spirit/category/admin/forms.py

```diff
@@ -36,3 +37,11 @@ def clean_parent(self):
                                               "can not have a parent since it has childrens"))
 
         return parent
+
+    def clean_colour(self):
+        colour = self.cleaned_data["colour"]
+
+        if colour != '' and not re.match(r'^#[a-f0-9]{3}([a-f0-9]{3}){0,1}$', colour):
```

Comment:
* There is no need for checking empty string, `if color ...` will do.
* The regex should match upper-case letters as well.